plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/procedure/FlushHiveMetastoreCacheProcedure.java

```diff
@@ -87,20 +93,46 @@ public Procedure get()
                         new Procedure.Argument("$FAKE_FIRST_PARAMETER", VARCHAR, false, FAKE_PARAM_DEFAULT_VALUE),
                         new Procedure.Argument(PARAM_SCHEMA_NAME, VARCHAR, false, null),
                         new Procedure.Argument(PARAM_TABLE_NAME, VARCHAR, false, null),
+                        new Procedure.Argument(PARAM_PARTITION_COLUMNS, new ArrayType(VARCHAR), false, null),
+                        new Procedure.Argument(PARAM_PARTITION_VALUES, new ArrayType(VARCHAR), false, null),
                         new Procedure.Argument(PARAM_PARTITION_COLUMN, new ArrayType(VARCHAR), false, null),
                         new Procedure.Argument(PARAM_PARTITION_VALUE, new ArrayType(VARCHAR), false, null)),
                 FLUSH_HIVE_METASTORE_CACHE.bindTo(this));
     }
 
-    public void flushMetadataCache(String fakeParam, String schemaName, String tableName, List<String> partitionColumn, List<String> partitionValue)
+    public void flushMetadataCache(
+            String fakeParam,
+            String schemaName,
+            String tableName,
+            List<String> partitionColumns,
+            List<String> partitionValues,
+            List<String> partitionColumn,
+            List<String> partitionValue)
     {
         checkState(FAKE_PARAM_DEFAULT_VALUE.equals(fakeParam), "Procedure should only be invoked with named parameters. " + PROCEDURE_USAGE_EXAMPLES);
+
+        Optional<List<String>> optionalPartitionColumns = Optional.ofNullable(partitionColumns);
+        Optional<List<String>> optionalPartitionValues = Optional.ofNullable(partitionValues);
+        Optional<List<String>> optionalPartitionColumn = Optional.ofNullable(partitionColumn);
+        Optional<List<String>> optionalPartitionValue = Optional.ofNullable(partitionValue);
+        checkState(((optionalPartitionColumns.isPresent() || optionalPartitionValues.isPresent())
+                        && optionalPartitionColumn.isEmpty() && optionalPartitionValue.isEmpty())
+                        || ((optionalPartitionColumn.isPresent() || optionalPartitionValue.isPresent())
+                        && optionalPartitionColumns.isEmpty() && optionalPartitionValues.isEmpty())
+                        || (optionalPartitionColumns.isEmpty() && optionalPartitionValues.isEmpty()
+                        && optionalPartitionColumn.isEmpty() && optionalPartitionValue.isEmpty()),
```

Comment:
I have no idea what this condition does :) Can it be refactored for readability?

Reply:
@ksobolew In general it enforces to use either one pair of params or another: partition_column, partition_value or partition_column[s], partition_value[s]. They should not be mixed.
I could potentially split it to 3 separate checkState calls. WDYT ?

Reply:
I mean, I "know" what it's trying to achieve, but looking at it I give up understanding it after the second nested `&&` and `||` or so ;)

Reply:
That's much better, thanks :) If I was really pedantic, I would ask for all these cases to be included in tests. Maybe it's overkill, though :)

Reply:
Actually they are already covered:
- testFlushPartitionCacheWithDeprecatedPartitionParams
- testFlushPartitionCache
- and 1 negative scenario in testIllegalFlushHiveMetastoreCacheProcedureCalls